src/App.tsx

```diff
@@ -1,28 +1,71 @@
-import React, { useState } from 'react';
+import React, { SetStateAction, useState } from 'react';
 import './App.css';
 
-const Image = (props: {rotation: number}) =>
-  <img
+type State = {
+  imageIndexOffset: number
+  imagesWithRotations: { url: string, rotation: number}[]
+}
+
+function Image(props: {state: State}){
+  const currentImage = props.state.imagesWithRotations[getCurrentImageIndex(props.state)];
```

Comment:
recommend only putting what's required in the `props` in this case the current image